### PR TITLE
Fixed range of ranks in Restrict Rank, see issue #434

### PR DESCRIPTION
**Pull request by IDex**
This small change fixes the issue in #434. The change doesn't really address why user.ranking is wrong in the first place though.

### CODE REVIEW

**Comment by BarneyStratford** [comment]
I'm not sure this is going to work as it stands, as professional players have ranks assigned to them rather than calculated from their rating. You're quite right though that the ranks that the backend returns to us are no longer being updated for amateur players, and hence data.get("user") can have the wrong rank.

This bug is actually already fixed upstream of here in the player_types PR that I've made.

**Comment by IDex** [comment]
Nice catch. Given the bug makes custom games nearly unplayable for any new, non-DDK players it would be nice to see it get fixed asap. I'm wondering if augmenting this quick fix is worth it/wanted, or is your fix going to get merged soon?

**Comment by BarneyStratford** [comment]
At the moment, I'm not sure if or when my PR will be accepted as there are some differences of opinion over it. If it is accepted, then it would be better to leave this part of the code as it is; otherwise, it's better to fix it here for now.

**Comment by anoek** [comment]
Sorry for the delay in reviewing this.

There's a few other considerations beyond just looking at a raw overall.rating -> rank conversion, so the preferred way to get a player's rank is to use the `getUserRating` method from `rank_utils`. In this case, you'd want to work with the `bounded_rank` which deals with appropriately bounding the rank between `25k` and `9d`. This method also deals with all of the pro/non-pro stuff as well.

So in short, do something like `let rank = getUserRating(data.get('user')).bounded_rank` and work from there. 

Thanks for the catch and the patch!

**Comment by anoek** [comment]
Awesome, thanks!
